corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -461,30 +463,36 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
         },
 
         addAddressPin: function (geocoder, addressMap, headers, model, addressIndex) {
-            let address = model.attributes.data[addressIndex];
-            geocoder.query(address, function (err, data) {
-                if (err === null) {
-                    let popupText = "";
-                    headers.forEach((header, index) => {
-                        if (header) {
-                            popupText += "<b>" + header + ":</b> " + model.attributes.data[index] + "<br>";
-                        }
-                    });
-                    L.marker(data.latlng)
-                        .addTo(addressMap)
-                        .bindPopup(popupText);
-                }
+            const address = model.attributes.data[addressIndex];
+            return new Promise((resolve, reject) => {
+                geocoder.query(address, function (err, data) {
```

Comment:
Just realized I don't understand what this is doing. `address` should already be a lat/long, why do we need to query mapbox?

Reply:
@orangejenny I was missing that context then. I thought it was a text field containing an address. That makes the code so much easier. Will change it to expect coordinates here